cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java

```diff
@@ -1411,6 +1416,34 @@ public void testDefaultImagePullSecrets() {
         assertEquals(0, ss.getSpec().getTemplate().getSpec().getImagePullSecrets().size());
     }
 
+    @Test
+    public void testSaslCluster() {
+        KafkaListenerAuthenticationSslPlaintextVault sslPlaintext = new KafkaListenerAuthenticationSslPlaintextVault();
+        String vaultAddr = "https://localhost:8200";
+        String vaultToken = "root-token";
+        sslPlaintext.setVaultAddr(vaultAddr);
+        sslPlaintext.setVaultToken(vaultToken);
+        Kafka kafkaAssembly = new KafkaBuilder(ResourceUtils.createKafkaCluster(namespace, cluster, replicas,
+                image, healthDelay, healthTimeout, metricsCm, configuration, emptyMap()))
+                .editSpec()
+                    .editKafka()
+                        .withNewListeners()
+                            .withNewPlain()
+                                .withKafkaListenerAuthenticationSslPlaintextVault(sslPlaintext)
+                            .endPlain()
+                        .endListeners()
+                    .endKafka()
+                .endSpec()
+                .build();
+        KafkaCluster kc = KafkaCluster.fromCrd(kafkaAssembly, VERSIONS);
+        List<EnvVar> envVars = kc.getEnvVars();
+        KafkaListeners listeners = kc.getListeners();
+        assertThat(envVars, hasItems(
```

Comment:
Is this really worth adding new dependency? It seems like something what can be easily implemented without Hamcrest. I would probably prefer to use the same as is used in all the other tests around.